pkg/apis/pipeline/v1/param_types.go

```diff
@@ -528,12 +528,22 @@ func validatePipelineParametersVariablesInTaskParameters(params []Param, prefix
 	return errs
 }
 
-// validatePipelineParametersVariablesInMatrixParameters validates matrix param value
+// validatePipelineParametersVariablesInMatrixParameters validates all pipeline paramater variables including Matrix.Params and Matrix.Include.Params
 // that may contain the reference(s) to other params to make sure those references are used appropriately.
-func validatePipelineParametersVariablesInMatrixParameters(matrix []Param, prefix string, paramNames sets.String, arrayParamNames sets.String, objectParamNameKeys map[string][]string) (errs *apis.FieldError) {
-	for _, param := range matrix {
-		for idx, arrayElement := range param.Value.ArrayVal {
-			errs = errs.Also(validateArrayVariable(arrayElement, prefix, paramNames, arrayParamNames, objectParamNameKeys).ViaFieldIndex("value", idx).ViaFieldKey("matrix", param.Name))
+func (matrix *Matrix) validatePipelineParametersVariablesInMatrixParameters(prefix string, paramNames sets.String, arrayParamNames sets.String, objectParamNameKeys map[string][]string) (errs *apis.FieldError) {
```

Comment:
can we please move this function to `pkg/apis/pipeline/v1/matrix_types.go`?

Reply:
along with the other `matrix` members: https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1/matrix_types.go

Reply:
Yes! Moved both matrix functions